features/kafka/producer/src/main/java/org/opennms/features/kafka/producer/collection/KafkaPersisterFactory.java

```diff
@@ -68,22 +71,50 @@ public Persister createPersister(ServiceParameters params, RrdRepository reposit
 
     public void init() throws IOException {
         // Create the Kafka producer
+        String metricsPid = KafkaProducerManager.METRICS_KAFKA_CLIENT_PID;
+        String globalPid = OpennmsKafkaProducer.KAFKA_CLIENT_PID;
+
         final Properties producerConfig = new Properties();
-        final Dictionary<String, Object> properties = configAdmin
-                .getConfiguration(OpennmsKafkaProducer.KAFKA_CLIENT_PID).getProperties();
-        if (properties != null) {
+        Dictionary<String, Object> properties = null;
+            properties = configAdmin.getConfiguration(metricsPid).getProperties();
+            if (hasValidConfiguration(properties)) {
+                LOG.info("Using metrics-specific Kafka configuration");
+            }
+
+
+        if (properties == null || properties.isEmpty() || properties.size() == 0) {
+            LOG.debug("Using global Kafka configuration for metrics");
+            properties = configAdmin.getConfiguration(globalPid).getProperties();
+        } else {
+            LOG.debug("Using metrics-specific Kafka configuration");
+        }
```

Comment:
This can be simplified to 

```
Dictionary<String, Object> properties = null;

      // Try metrics-specific first
      properties = configAdmin.getConfiguration(metricsPid).getProperties();
      if (hasValidConfiguration(properties)) {
          LOG.info("Using metrics-specific Kafka configuration");
      } else {
          // Fall back to global
          LOG.debug("Metrics-specific config not valid, falling back to global");
          properties = configAdmin.getConfiguration(globalPid).getProperties();
      }
```